src/routers/handlers/personal-code/personalCodeHandler.ts

```diff
@@ -100,14 +101,26 @@ export class PersonalCodeHandler extends GenericHandler<PersonalCodeViewData> {
     }
 
     private resolveNextPageUrl (validationStatusResponse: Resource<ValidationStatusResponse>): string {
-        let url = PrefixedUrls.INDIVIDUAL_STATEMENT;
+        const url: string = PrefixedUrls.INDIVIDUAL_STATEMENT;
 
         if (validationStatusResponse.resource && validationStatusResponse.resource.isValid === false) {
-            const hasNameMismatchError = validationStatusResponse.resource.errors?.some((validationError: { error: string | string[]; }) => validationError.error.includes("name mismatch"));
-            url = hasNameMismatchError ? PrefixedUrls.NAME_MISMATCH : PrefixedUrls.INDIVIDUAL_STATEMENT;
+            if (this.hasError(getDobValidationMessage(), validationStatusResponse.resource.errors) ||
+                this.hasError(getUvidValidationMessages(), validationStatusResponse.resource.errors)) {
+                return getUrlWithStopType(toStopScreenPrefixedUrl(STOP_TYPE.PSC_DOB_MISMATCH), STOP_TYPE.PSC_DOB_MISMATCH);
```

Comment:
would it be neater to write
`return getUrlWithStopType(PrefixedUrls.STOP_SCREEN, STOP_TYPE.PSC_DOB_MISMATCH)`